ui/plugin-system/src/runtime/template-variables.ts

```diff
@@ -21,9 +22,106 @@ export type VariableState = {
   options?: VariableOption[];
   loading: boolean;
   error?: Error;
+  /**
+   * If a local variable is shadowing an external variable, local var will have the flag ``shadowing=true``.
+   */
+  shadowing?: boolean;
```

Comment:
I'm not sure that I 100% understand the `shadowing` concept, is it similar to what @saminzadeh created `dependsOn` for?

Reply:
Trying to use a project variable in a dashboard variable, it seems to work in the variable list drawer, but on a dashboard, the panels show No Data (assuming related to the loading order of the variables), also when I go to save I get:

<img width="1707" alt="image" src="https://github.com/perses/perses/assets/9369625/63474f8b-6e98-4d55-ab5c-3b386636d24d">

Full recording [mp4 link here](https://drive.google.com/file/d/1-Sy1WbvSMZe7nAhcSb_G3Kf5OOGclY1V/view?usp=drive_link) (was too large to upload)

Reply:
Shadowing means that the variable definition is used in place of another of the same name but with lower priority. (i.e. a local variable with name interval will shadow/override any external variable with that name)

We could replace 'shadowing' by 'overriding' if it's clearer. The word is coming from the datasources design ping @AntoineThebaud

Reply:
I will check the behavior that you described in detail. There's maybe a bug in my implementation

Reply:
Yes regarding the shadowing/shadowed behavior you have info on the Datasources design discussion https://github.com/perses/perses/discussions/404

Just in case, other ideas to name this:
- masking/masked
- overriding/overridden